### ISSUE TITLE
Support for partial match

### ISSUE THREAD

**Issue by mrkkrp** [yesodweb/persistent#584]
I would like to use SQL feature that allows to enter just part of string to search, like:

```sql
SELECT * FROM my_table WHERE first_name LIKE '%foo%';
```

I don't see how to do it with Persistent.

**Comment by psibi** [comment]
Did you try using `rawSql` ?

**Comment by mrkkrp** [comment]
`rawSql` should not really be the way to do such common things, don't you think? Every time I'm trying to use `rawSql` it feels sort of painful and cumbersome experience.

**Comment by psibi** [comment]
Yeah. But I don't think writing a `rawSql` based function is that cumbersome, for example:

```
getPersonBlog :: MonadIO m => ReaderT SqlBackend m [(Entity Person, Entity BlogPost)]
getPersonBlog = rawSql "select ??,?? from person,blog_post where person.id = blog_post.author_id" []
```

Maybe a utility function for this can be added up in `persistent`.

**Comment by snoyberg** [comment]
It's possible to do this with BackendSpecificFilter (
https://www.stackage.org/haddock/lts-6.5/persistent-2.2.4.1/Database-Persist-Types.html#v:BackendSpecificFilter),
though I haven't tried it myself in a while.

**Comment by gregwebs** [comment]
Should be backend specific because semantics of these are always a different across backends  MongoDB has a lot of specific filters already. Esqueleto also supports this operation

**Comment by mrkkrp** [comment]
@snoyberg, `BackendSpecificFilter` looks promising for my use case. Is there an example of its use somewhere?

**Comment by psibi** [comment]
This may help you: https://stackoverflow.com/questions/11048143/example-of-persistent-with-backend-specific-operator

**Comment by snoyberg** [comment]
I don't have one handy, perhaps there's one on StackOverflow.

**Comment by snoyberg** [comment]
Hah, Sibi to the rescue :)
